lib/mongo/error/operation_failure.rb

```diff
@@ -182,6 +182,16 @@ def initialize(message = nil, result = nil, options = {})
         @code = options[:code]
         @code_name = options[:code_name]
         super(message)
+
+        if result
+          err_doc = result.send(:first_document)
+
+          if err_doc && err_doc['errorLabels']
+            err_doc['errorLabels'].each do |label|
+              add_label(label)
+            end
```

Comment:
There is a Mongo::Error::Parser which parses all the other fields out of the result, how come this code is added to the exception class and not to the parser?

Reply:
Good idea! I'll change it to do that